src/common/cockpitpipe.c

```diff
@@ -440,6 +441,9 @@ set_problem_from_errno (CockpitPipe *self,
     problem = "access-denied";
   else if (errn == ENOENT || errn == ECONNREFUSED)
     problem = "not-found";
+  /* only warn about Cockpit-internal fds, not opaque user ones */
+  else if (errn == EBADF && self->priv->is_user_fd)
+    problem = "internal-error";
```

Comment:
Is this really an "internal-error"? That signifies an internal failure in Cockpit. Do you think another error code better applies? Perhaps "protocol-error" which denotes that the protocol was used wrong? Lets not invent a new code, but I think we may be able to find a better one among the already existing problem codes.